gui/tools/replication_adapter.py

```diff
@@ -0,0 +1,15 @@
+from types import SimpleNamespace
+
+import requests
+
+from freenasUI.middleware.client import client
+
+
+def query_model(path, filter_key):
+    with client as c:
+        token = c.call("auth.generate_token")
+
+    r = requests.get(f"http://localhost/api/v1.0/{path}/", headers={"Authorization": f"Token {token}"})
```

Comment:
There is no guarantee we will be listening at localhost in standard port.

Perhaps we have to connect directly to django fastcgi?

Reply:
Can we detect which address we are listening and connect to it instead? I don't like the idea of having a FastCGI client here.

Reply:
Well, using HTTP will be more error prone but I suppose we can use the ui port attribute from system.general.config

Also in 11.3 we are defaulting to https, so you are gonna have to use verify=False also